pkg/worker/worker.go

```diff
@@ -494,10 +496,22 @@ func (w *Worker) printBuildInfo() {
 func (w *Worker) Run() (bool, error) {
 	var success bool
 	w.initFilterFunc()
-	err := w.checkForNewerBuilds()
+	exists, err := w.checkForNewerBuilds()
 	if err != nil {
 		return false, err
 	}
+	if exists {
+		return false, fmt.Errorf("newer build found, not moving forward")
+	}
+	go func() {
+		for {
+			exists1, _ := w.checkForNewerBuilds()
+			if exists1 {
+				log.Fatalf("found newer build, cancelling current build")
```

Comment:
So it will simply exit out from the current build?

Reply:
Yep. 

The directory created for testing should get cleaned up by the newer job in the queue and any other cleanup is responsibility of test scripts (which should happen anyway as job can get preempted for other reasons so your script should cleanup stuff created by previous run)

Reply:
lgtm